sunpy/data/manager/manager.py

```diff
@@ -0,0 +1,135 @@
+import pathlib
+import functools
+from contextlib import contextmanager
+
+from sunpy.util.util import hash_file
+
+
+class DataManager:
+    """
+    DataManager
+    """
+
+    def __init__(self, cache):
+        # TODO: Folder prefix should be init argument
+        self._cache = cache
+
+        self._file_cache = {}
+
+        self._skip_hash_check = False
+        self._skip_file = {}  # Dict[str, str]
+
+    def require(self, name, urls, sha_hash):
+        """decorator for doing stuff
+
+        Parameters
+        ----------
+        name: str
+        The name to reference the file with
+        urls: list
+        List of urls to download the file from
+        sha_hash: str
+        Hash of file
+        """
+        def decorator(func):
+            @functools.wraps(func)
+            def wrapper(*args, **kwargs):
+                # TODO: Refractor into function(s)
```

Comment:
You mean turn this into a series of functions instead of this one large block of code?

Reply:
I guess.

Reply:
Well if you dont want to, you don't have to. You can just remove the todo!